### DISCUSSION TITLE
How to write schema such that i get a proper array without descriptors and brackets in dynamooose

### DISCUSSION THREAD

**Discussion by sagar-jg** [dynamoose/dynamoose#1185] [Q&A]
I am inserting an array to the DB in this format from postman -> 

```
{
     "cat_id": ["1","3"]
}
```

I have a model as follows
```

var dynamoose = require('dynamoose');

var agentSchema = new dynamoose.Schema({
    agent_id : {type : String,  hashKey: true},
    category_id  : {type: Array, schema :[String]},
},
{
    timestamps:true
}
)
module.exports = dynamoose.model('agents_wa1', agentSchema)

```

But when i check in DB, it is stored in this form

```
 "category_id": {
                "L": [
                    {
                        "S": "1"
                    },
                    {
                        "S": "3"
                    }
                ]
            },
```

Instead i wanted it like :   ["1", "3"]   in the proper array form, instead of those brackets
Because i get an error in another api when I get category_id in the above format.(in code)

**Comment by fishcharlie** [comment]
@sagar-jg That is the expected result. DynamoDB has a very specific format for storing data. [Dynobase](https://dynobase.dev/?ref=dynamoose) has a tool for visualizing this: https://dynobase.dev/dynamodb-json-converter-tool/. If you were to also look at this in Dynobase or any console, it will show your expected result.

There is no way to store it as `["1","3"]` directly in DynamoDB. Feel free to try it on your own, but any tool will convert it as DynamoDB will throw an error otherwise.

So you will have to use Dynamoose, another tool, or convert it to your expected format yourself.

**Reply by sagar-jg** [reply]
Ok, Thank you.
I tried using type Set, and was getting in that format, but instead i needed same thing for Array.
Anyways thanks for the response

**Reply by sagar-jg** [reply]
@fishcharlie, When i write scheme like this,

```
category_id : {
        type: Array,
        required : true
    }
```

The array data that i pass is not stored in DB, it gives me

`category_id : [ ]`


This is the way im saving it to DB

```
let agent = new Agent();
        agent.agent_id = uuidv4();
        agent.category_id = data.cat_id;
        console.log(Array.isArray(agent.category_id))
        agent.agent_name = data.agent_name;
        agent.agent_language = data.agent_lang;



        console.log(JSON.stringify(agent));

        agent.save(function (err, result) {
            if (err) {
                console.log(err);
                callback({ "statuscode": 500, "message": "Error while saving data" });
                return;
            }
            console.log(JSON.stringify(result));
            callback({ "statuscode": 200, "message": "Saved successfully" });
            return;
        })
```
